### DISCUSSION TITLE
feature-request: table-panel - numbered rows

### DISCUSSION THREAD

**Discussion by pfischer64** [grafana/grafana#38171] [Dashboards, Panels & Visualizations]
When I use elasticsearch as a datasource, there is no easy solution to display numbered rows in a table panel. 

Therefore I suggest a new table option:

Show Row Numbers - toggle to show/hide row numbers

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by nemani** [comment]
Any update on this?

**Comment by xiaopeng163** [comment]
we also need this feature

**Comment by relishcolouredhat** [comment]
Agreed that this is a useful + missing feature that would greatly enhance the builtin table plugin's utility; removing the need to use community table plugins in my case.

Row numbers are great for communication & when combined with sorting at the datasource; they allow for easily providing a 'rank' for each object relative to it's peers. 

This is especially great when building 'scoreboards'; or in cases where a table holds a very large number of rows, especially when paginated. 

##### Use case example
- when given a query which returns a table that has been sorted by the datsource;
- and enabled by the dashboard editor;
- grafana would number the lines based upon the order presented by the datasource;
- the line number should remain tied the to the entries in each row.
- This now allows other sorts on the same table to be compared against the original order, AND/OR
- allows comparison of relative position within the query without needing to include further context ( eg info from one line plus footer can communicate "container 1234 is the 5th worst container in our infra- it's worse than 6789 other containers" ).

##### Specific example
- Given a query returning a table with the following three columns, which has been sorted by the datasource on the value of 'memory allocation';
    - Column 1: string representing container name,
    - Column 2: numeric value representing memory allocation
    - Column 3: numeric value representing cpu load
- The dashboard editor would enable 'row numbering', causing a 4th column to be created, containing an index of the row, prior to any column sort applied in the grafana ux
- A dashboard user can now:
    - provide a 'rank' for any given container, relative to the 'best' and 'worst' container; simply by searching for it's name/id and comparing the row number. 
    - use column sorting on other columns, within the standard table plugin, to compare "containers by cpu" vs their "order" in the "list of top containers by memory"; 
  
Thank you kindly for your consideration.

**Reply by NargiT** [reply]
what community plugin do you use ?

**Comment by NargiT** [comment]
Definitely need this as an option

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/dataviz-squad

**Comment by Falmarri** [comment]
@natellium do I have this right that this was originally an issue, was moved to a discussion, then closed because you're moving back to issues? Why not just re-open the original issue then? https://github.com/grafana/grafana/issues/28703

**Comment by Funecio-Agrante** [comment]
I also agree this feature is necessary. And there is already something close that only needs to start counting from 1 instead of zero. Should be a straightforward feature to add?...
In 'Transform Data' > 'Add field from calculation' there is a 'Row index' Mode, but this unsurprisingly starts counting at 0, instead of 1.

When I need a rank field in Sqlite, I add something like 
`SELECT ROW_NUMBER() OVER (ORDER BY COUNT(*) DESC) AS Rank` or
`SELECT ROW_NUMBER() OVER (ORDER BY field_name ASC) AS Rank` depending on the situation. I suppose some times the data isn't generated from SQL queries inside Grafana, but maybe in those cases two JOINed queries would solve the issue?
